services/src/main/java/org/fao/geonet/api/registries/vocabularies/KeywordsApi.java

```diff
@@ -382,6 +379,112 @@ public Object getKeywordById(
         @Parameter(hidden = true)
         HttpServletRequest request
         ) throws Exception {
+        return getKeyword(uri,sThesaurusName,langs, keywordOnly, transformation,langMapJson,allRequestParams, accept, request);
+    }
+
+    /**
+     * Gets the keyword by id.
+     *
+     * @param uri              the uri
+     * @param sThesaurusName   the s thesaurus name
+     * @param langs            the langs
+     * @param keywordOnly      the keyword only
+     * @param transformation   the transformation
+     * @param allRequestParams the all request params
+     * @param request          the request
+     * @return the keyword by id
+     * @throws Exception the exception
+     */
+    @io.swagger.v3.oas.annotations.Operation(
+        summary = "Get keyword by id",
+        description = "Retrieve XML representation of keyword(s) from same thesaurus" +
+            "using different transformations. " +
+            "'to-iso19139-keyword' is the default and return an ISO19139 snippet." +
+            "'to-iso19139-keyword-as-xlink' return an XLinked element. Custom transformation " +
+            "can be create on a per schema basis." +
+        "Use POST method if you want to support larger parameters list"
+    )
+    @RequestMapping(
+        path = "/keywordById",
```

Comment:
Better to keep GET unchanged because it is used in all records when XLink mode is enabled in a catalogue (so it will avoid migration in metadata records). And we can change POST (with `keywordByIds`) and its (probably only) usage in https://github.com/geonetwork/core-geonetwork/blob/main/web-ui/src/main/resources/catalog/components/utility/UtilityService.js#L709-L710. Then it looks good. @josegar74 can you think of any other usages of this ?

Reply:
@fxprunayre 
Sorry - I had accidently changed the endpoint to /keywordById.  I reverted it back to /keyword for both get and post.